FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/mbedtls_freertos.c

```diff
@@ -0,0 +1,541 @@
+    mbedtls_pk_init( &( pSslContext->privKey ) );
+    mbedtls_x509_crt_init( &( pSslContext->clientCert ) );
+    mbedtls_ssl_init( &( pSslContext->context ) );
+}
+/*-----------------------------------------------------------*/
+
+static void sslContextFree( SSLContext_t * pSslContext )
+{
+    mbedtls_ssl_free( &( pSslContext->context ) );
+    mbedtls_x509_crt_free( &( pSslContext->rootCa ) );
+    mbedtls_x509_crt_free( &( pSslContext->clientCert ) );
+    mbedtls_pk_free( &( pSslContext->privKey ) );
+    mbedtls_ssl_config_free( &( pSslContext->config ) );
+}
+
+/*-----------------------------------------------------------*/
+
+static MbedtlsStatus_t tlsSetup( NetworkContext_t * pNetworkContext,
+                                 const char * pHostName,
+                                 const NetworkCredentials_t * pNetworkCredentials )
+{
+    MbedtlsStatus_t returnStatus = MBEDTLS_SUCCESS;
+    int mbedtlsError = 0;
+
+    /* Initialize the mbed TLS context structures. */
+    sslContextInit( &( pNetworkContext->sslContext ) );
+
+    mbedtlsError = mbedtls_ssl_config_defaults( &( pNetworkContext->sslContext.config ),
+                                                MBEDTLS_SSL_IS_CLIENT,
+                                                MBEDTLS_SSL_TRANSPORT_STREAM,
+                                                MBEDTLS_SSL_PRESET_DEFAULT );
+
+    if( mbedtlsError != 0 )
+    {
+        LogError( ( "Failed to set default SSL configuration, error %d.", mbedtlsError ) );
+
+        /* Per mbed TLS docs, mbedtls_ssl_config_defaults only fails on memory allocation. */
+        returnStatus = MBEDTLS_INSUFFICIENT_MEMORY;
+    }
+
+    if( returnStatus == MBEDTLS_SUCCESS )
+    {
+        /* Set up the certificate security profile, starting from the default value. */
+        pNetworkContext->sslContext.certProfile = mbedtls_x509_crt_profile_default;
+
+        /* test.mosquitto.org only provides a 1024-bit RSA certificate, which is
+         * not acceptable by the default mbed TLS certificate security profile.
+         * For the purposes of this demo, allow the use of 1024-bit RSA certificates.
+         * This block should be removed otherwise. */
+        if( strncmp( pHostName, "test.mosquitto.org", strlen( pHostName ) ) == 0 )
+        {
+            pNetworkContext->sslContext.certProfile.rsa_min_bitlen = 1024;
+        }
+
+        /* Set SSL authmode and the RNG context. */
+        mbedtls_ssl_conf_authmode( &( pNetworkContext->sslContext.config ),
+                                   MBEDTLS_SSL_VERIFY_REQUIRED );
+        mbedtls_ssl_conf_rng( &( pNetworkContext->sslContext.config ),
+                              mbedtls_ctr_drbg_random,
+                              &ctrDrgbContext );
+        mbedtls_ssl_conf_cert_profile( &( pNetworkContext->sslContext.config ),
+                                       &( pNetworkContext->sslContext.certProfile ) );
+    }
+
+    if( returnStatus == MBEDTLS_SUCCESS )
```

Comment:
This if-block can be merged into the above one as there is no update to the `returnStatus` in the above if-block